### PR TITLE
Add kubelet to kubernetes-master

### PR DESCRIPTION
**Pull request by Cynerva**
Marked as draft until we can safely merge this without disrupting the 1.23 release.

This will need to be merged together with:
* https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/103
* https://github.com/charmed-kubernetes/layer-kubernetes-common/pull/21
* https://github.com/charmed-kubernetes/layer-kubernetes-master-worker-base/pull/13

### CODE REVIEW

**Comment by Cynerva** [comment]
Rebased and ready for review now that we can safely merge to master.

File: reactive/kubernetes_master.py [hunk 1]
```diff
@@ -3546,3 +3581,71 @@ def send_default_cni():
...
+
+    registry = hookenv.config("image-registry")
+    taints = hookenv.config("register-with-taints").split()
+    kubernetes_common.configure_kubelet(dns_domain, dns_ip, registry, taints=taints)
+    service_restart("snap.kubelet.daemon")
+
+    set_flag("kubernetes-master.kubelet.configured")
+
+
+@when_any("config.changed.kubelet-extra-args", "config.changed.kubelet-extra-config")
```
Comment (addyess) [inline reactive/kubernetes_master.py:3624]: Are you planning another PR with `kubelet-extra-args` or `kubelet-extra-config` in the `config.yaml`?

Comment (addyess) [reply reactive/kubernetes_master.py:3624]: i think the `layer-kubernetes-common` looks up that config.  This is probably the right time to pull those config items from `kubernetes-worker`

Comment (Cynerva) [reply reactive/kubernetes_master.py:3624]: Yup, a PR for this already exists, but it is targeted against a base layer that's shared by both kubernetes-master and kubernetes-worker: https://github.com/charmed-kubernetes/layer-kubernetes-master-worker-base/pull/13

Comment (addyess) [reply reactive/kubernetes_master.py:3624]: 🤦🏻‍♂️ oh there they are!  All's well

**Review by addyess — APPROVED** [review]

**Pull request merged**